workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityReaderWriterGenerator.tt

```diff
@@ -27,9 +27,9 @@ namespace <#= qualifiedNamespace #>
             [InjectableId(InjectableType.ReaderWriter, <#= unityComponentDefinition.Id #>)]
             internal class ReaderWriterCreator : IInjectableCreator
             {
-                public IInjectable CreateInjectable(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher)
+                public IInjectable CreateInjectable(Entity entity, World world, ILogDispatcher logDispatcher)
```

Comment:
after looking through all the files, it doesn't seem like the world is even needed? why was this argument added in the first place instead. it's only used to create the entity manager now

Reply:
It is needed for the `WorldCommandResponseHandler` to register itself into the `GameObjectWorldCommandSystem` for that world in order to be able to be linked to the entity that owns it.

```cs

            private WorldCommandResponseHandler(Entity entity,
                World world,
                ILogDispatcher logDispatcher)
            {
                world.GetOrCreateManager<GameObjectWorldCommandSystem>().RegisterResponseHandler(entity, this);

                this.entity = entity;
                this.logDispatcher = logDispatcher;
            }
```

If we can come up with a better way to do this I'm happy to adjust the code.

Reply:
Where is that though? In this PR I can't find a CreateInjectable that uses it.

Reply:
https://github.com/spatialos/UnityGDK/pull/301/files#diff-3e458a8eb1c1fb7860fe38a3c05a536dR97